Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +352,873 @@ public override bool IsServerConnected()
+                            {
+                                groupIndx++;
+                                if (string.IsNullOrEmpty(ident.ActivityExternalID) ||
+                                        tc.Steps.Where(x => x.StepID == ident.ActivityExternalID).FirstOrDefault() == null)
+                                {
+                                    continue;//activity which not originally came from the TC
+                                }
+                                numOfSeenSteps++;
+
+                                if (numOfSeenSteps >= stepIndx) { break; }
+                            }
+                            ActivityIdentifiers identOnPlace = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers[groupIndx];
+                            if (identOnPlace.ActivityGuid != act.Guid)
+                            {
+                                //replace places in group
+                                tcActivsGroup.ActivitiesIdentifiers.Move(groupActIndx, groupIndx);
+                                //replace places in business flow
+                                busFlow.Activities.Move(bfActIndx, startGroupActsIndxInBf + groupIndx);
+                            }
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex);
+                        //failed to re order the activities to match the tc steps order, not worth breaking the import because of this
+                    }
+                }
+
+                //Add the BF variables (linked variables)
+                if (busVariables.Keys.Count > 0)
+                {
+                    foreach (KeyValuePair<string, string> var in busVariables)
+                    {
+                        //add as String param
+                        VariableString busVar = new VariableString();
+                        busVar.Name = var.Key;
+                        busVar.InitialStringValue = var.Value;
+                        busFlow.AddVariable(busVar);
+                    }
+                }
+
+                return busFlow;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to import QC test set and convert it into " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow), ex);
+                return null;
+            }
+        }
+
+        public bool ExportBusinessFlow(BusinessFlow businessFlow, QCTestSet mappedTestSet, string fatherId, ObservableList<ExternalItemFieldBase> testSetFields, ObservableList<ExternalItemFieldBase> testInstanceFields, ref string result)
+        {
+            int testSetId = 0;
+            // ObservableList<ActivitiesGroup> existingActivitiesGroups = new ObservableList<ActivitiesGroup>();
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)